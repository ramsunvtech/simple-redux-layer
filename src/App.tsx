import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import CreatePart from './pages/CreatePart';
import Home from './pages/Home';
import Header from './components/Header';

function App() {
  return (
    <Router>
      <div className="App">
        <Header />
        <div>
          <Route exact path="/" component={Home} />
          <Route path="/create" component={CreatePart} />
        </div>
      </div>
    </Router>
  );
}

export default App;
