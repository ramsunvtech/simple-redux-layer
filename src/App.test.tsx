import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';

// `Test` Component.
import App from './App';

// Mock the Home Page.
jest.mock('./pages/Home');

describe('App', () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
