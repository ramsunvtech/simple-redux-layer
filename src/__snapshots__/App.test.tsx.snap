// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      data-testid="header"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
          data-testid="title"
        >
          Simple Redux Layer
        </h6>
          -  
        <a
          class="makeStyles-topLink-1"
          href="/simple-redux-layer/"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            data-testid="home"
          >
            Home
          </h6>
        </a>
          |  
        <a
          class="makeStyles-topLink-1"
          href="/simple-redux-layer/create"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            data-testid="create"
          >
            Create Part
          </h6>
        </a>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
